### PR TITLE
WIP autoloader autogen + packages

### PR DESCRIPTION
**Pull request by ngroman**
### Test plan

See included automated tests.

### CODE REVIEW

File: test/cli/autogen-pkg-autoloader/autogen-pkg-autoloader.out [hunk 1]
```diff
@@ -140,6 +142,18 @@ require 'my_gem'
...
 
+--- output/RootPackage/Nested.rb
+# frozen_string_literal: true
+# typed: true
+
+Opus::Require.on_autoload('RootPackage::Nested')
+
+module RootPackage::Nested
+end
+Opus::Require.register_package(RootPackage::Nested, 'RootPackage::Nested')
```
Comment (ngroman) [inline test/cli/autogen-pkg-autoloader/autogen-pkg-autoloader.out:153]: Depending on what we want we could also make the 2nd arg something more like `RootPackage_Nested` or however we want to munge.

**Comment by ngroman** [comment]
cc @aadi-stripe

File: test/cli/autogen-pkg-autoloader/autogen-pkg-autoloader.out (hunk 1 above)
Comment (aadi-stripe) [reply test/cli/autogen-pkg-autoloader/autogen-pkg-autoloader.out:153]: Ah ok.

**Comment by ngroman** [comment]
Subsumed into https://github.com/sorbet/sorbet/pull/5216

**Pull request closed without merging**